omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/AdministrationLinksResource1_8.java

```diff
@@ -0,0 +1,180 @@
+		return null;
+	}
+
+	@Override
+	public ModuleAdministrationLinks1_8 getByUniqueId(String moduleId) {
+		// Assumes that moduleId is an id of a module that user wants to get admin links for.
+		// AdministrationLinksResource has no id by itself as it's not an OpenMRS data object.
+
+		ModuleAdministrationLinks1_8 administrationLinks = getAdministrationLinksForModule(moduleId);
+		if (administrationLinks == null) {
+			throw new ObjectNotFoundException();
+		}
+
+		return administrationLinks;
+	}
+
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		if (rep instanceof DefaultRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("display");
+			description.addProperty(MODULE_TITLE);
+			description.addProperty(LINKS);
+			description.addSelfLink();
+			return description;
+		} else if (rep instanceof FullRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("display");
+			description.addProperty(MODULE_TITLE);
+			description.addProperty(LINKS);
+			description.addSelfLink();
+			return description;
+		} else if (rep instanceof RefRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("display");
+			description.addProperty(MODULE_TITLE);
+			description.addProperty(LINKS);
+			description.addSelfLink();
+			return description;
+		}
+		return null;
+	}
+
+	@PropertyGetter("uuid")
+	public static String getUuid(ModuleAdministrationLinks1_8 instance) {
+		return instance.getModuleId();
+	}
+
+	@PropertyGetter("display")
+	public static String getDisplay(ModuleAdministrationLinks1_8 instance) {
+		return instance.getModuleTitle();
+	}
+
+	@Override
+	public Model getGETModel(Representation rep) {
+		ModelImpl model = (ModelImpl) super.getGETModel(rep);
+		if (rep instanceof DefaultRepresentation || rep instanceof FullRepresentation) {
+			model
+					.property(MODULE_TITLE, new StringProperty())
+					.property(LINKS, new ArrayProperty(new ObjectProperty()));
+		}
+
+		return model;
+	}
+
+	@Override
+	public NeedsPaging<ModuleAdministrationLinks1_8> doGetAll(RequestContext context) throws ResponseException {
+		return new NeedsPaging<ModuleAdministrationLinks1_8>(getAllAdministrationLinks(), context);
+	}
+
+	private ModuleAdministrationLinks1_8 getAdministrationLinksForModule(String moduleId) {
+		List<Extension> adminListsExtensions = moduleFactoryWrapper.getExtensions(ADMIN_LIST_POINT_ID);
+		for (Extension adminListExtension : adminListsExtensions) {
+			if (adminListExtension instanceof AdministrationSectionExt && adminListExtension.getModuleId()
+					.equals(moduleId)) {
+				return getModuleWithAdminLinks(adminListExtension);
```

Comment:
Does the above method end up calling moduleFactoryWrapper.getExtensions() along the way?

Reply:
It doesn't. By the way, it will be refactored since I'm going to use `AdministrationSectionExt` directly in response.

Reply:
ok let me wait for the refactoring.